File: php/mediator/President.php
<?php

class President implements Mediator
{
    private $map;

    public function register($dname, $d)
    {
        $this->map[$dname] = $d;
    }

    public function command($dname)
    {
        $d = $this->map[$dname];
        $d->selfAction();
    }
}

File: java/pattern/src/mediator/President.java
package mediator;

import java.util.HashMap;

public class President implements Mediator {
    private HashMap<String, Department> map = new HashMap<>();

    @Override
    public void register(String dname, Department d) {
        map.put(dname, d);
    }

    @Override
    public void command(String dname) {
        map.get(dname).selfAction();
    }
}
